### DISCUSSION TITLE
Custom Module Development

### DISCUSSION THREAD

**Discussion by techravi007** [oqtane/oqtane.framework#3688] [Q&A] [answered]
Can someone provide me reference for updated documentation / video-series  on custom module development ? 
This seems to be obsolete... [Getting Started with Oqtane Module Development](https://www.youtube.com/watch?v=JPfUZPlRRCE&list=PLYhXmd7yV0elLNLfQwZBUlM7ZSMYPTZ_f)

**Comment by thabaum** [comment]
Hello @techravi007 ,
[docs.oqtane.org guides](https://docs.oqtane.org/guides/index.html) is one source for some information that we are currently working on updating.

There will probably be some fresh articles coming in our new Wiki @sbwalker is working on producing for us now with an article to help take modules to the marketplace.  I am advocating for a learn.oqtane.com information hub so we can refer you there and showcase all videos and learning resources available.   

The meetup video for building themes with Material and Mud Blazor starring @markdav-is shows a lot of very valuable information and  discusses similarities to module creation that may clear something up.  It sounded like it is more difficult to do a theme than a module, so if you learn what is in that video it may clear up some of the current processes.  #3629 

#1324 also discusses some of the resources available to help assist.

I know the community will be providing more in the near future.  Oqtane is starting to really grab gears while staying hooked up to the road here!

Cheers!

**Comment by sbwalker** [comment, marked as answer]
@techravi007 the Module Development video series is from 3 years ago however all of the concepts demonstrated are still relevant. The module which the video series is based upon is open source and is located in this repo: https://github.com/oqtane/DNF.Projects. The module is running in production at https://dnfprojects.org. The only significant difference in the module code between the video series and current bits (other than the .NET version) is the Repository patterns, which now uses EF Core Migrations rather than SQL Scripts (however both approaches are still supported in the framework). 

Is there a particular reason why you indicated that the video series is obsolete?

**Reply by techravi007** [reply]
Thanks for the clarification @sbwalker. :pray:
Repository patterns is the major difference which I noticed and module created by latest module creator doesn't have explicit "Script" folder reference, which can state it supports sql scripts as well.
